cmds/dd/dd_test.go

```diff
@@ -160,18 +160,24 @@ func stdoutEqual(i io.Reader, o []byte, _ int64) error {
 // is returned. Otherwise an error is returned for a non-matching byte or if the count doesn't
 // match.
 func byteCount(i io.Reader, o []byte, n int64) error {
-	b := bufio.NewReader(i)
 	var count int64
+	buf := make([]byte, 4096)
 
 	for {
-		z, err := b.ReadByte()
-		if err != nil {
+		read, err := i.Read(buf)
+		if err != nil || read == 0 {
 			break
 		}
-		if z == o[0] {
-			count++
-		} else {
-			return fmt.Errorf("Found non-matching byte: %v, at offset: %d", o[0], count)
+		for z := 0; z < read; z++ {
+			if buf[z] == o[0] {
+				count++
+			} else {
+				return fmt.Errorf("Found non-matching byte: %v, at offset: %d", o[0], count)
```

Comment:
Not related to your PR, but if you don't mind, can you change the print to print buf[z] rather than o[0]. In the cases we use this, I think we are comparing everything to o[0], so it's more useful to know what the wrong value is rather than the target value. Or print both, that's also fine.

Reply:
oops, also corrected.

thanks for your comments @GanShun